tailnet/controllers.go

```diff
@@ -901,19 +910,52 @@ type agent struct {
 	name string
 }
 
-func (t *tunnelAllWorkspaceUpdatesController) New(client WorkspaceUpdatesClient) CloserWaiter {
+func (t *TunnelAllWorkspaceUpdatesController) New(client WorkspaceUpdatesClient) CloserWaiter {
+	t.mu.Lock()
+	defer t.mu.Unlock()
 	updater := &tunnelUpdater{
 		client:         client,
 		errChan:        make(chan error, 1),
 		logger:         t.logger,
 		coordCtrl:      t.coordCtrl,
 		dnsHostsSetter: t.dnsHostSetter,
+		updateHandler:  t.updateHandler,
 		ownerUsername:  t.ownerUsername,
 		recvLoopDone:   make(chan struct{}),
 		workspaces:     make(map[uuid.UUID]*workspace),
 	}
-	go updater.recvLoop()
-	return updater
+	t.updater = updater
+	go t.updater.recvLoop()
+	return t.updater
+}
+
+func (t *TunnelAllWorkspaceUpdatesController) CurrentState() *proto.WorkspaceUpdate {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	if t.updater == nil {
+		return nil
```

Comment:
Would it make sense to return `(*proto.WorkspaceUpdate, error)` here if `t.updater` is nil?